src/transformers/models/owlvit/modeling_owlvit.py

```diff
@@ -366,6 +366,64 @@ def forward(
         return embeddings
 
 
+def eager_attention_forward(
+    module: nn.Module,
+    query: torch.Tensor,
+    key: torch.Tensor,
+    value: torch.Tensor,
+    attention_mask: Optional[torch.Tensor],
+    scaling: float,
+    dropout: float = 0.0,
+    **kwargs,
+):
+    bsz, num_heads, seq_len, head_dim = query.shape
+    proj_shape = (bsz * num_heads, -1, head_dim)
+    query = query.reshape(proj_shape)
+    key = key.reshape(proj_shape)
+    value = value.reshape(proj_shape)
+
+    attn_weights = torch.bmm(query, key.transpose(1, 2)) * scaling
+
+    if attn_weights.size() != (bsz * num_heads, seq_len, seq_len):
+        raise ValueError(
+            f"Attention weights should be of size {(bsz * num_heads, seq_len, seq_len)}, but is {attn_weights.size()}"
+        )
+
+    if attention_mask is not None:
```

Comment:
Can we reuse one of the existing `eager_attention_forward` functions defined in transformers?

Reply:
Going through the code, I think the BLIP2 eager should achieve the same thing so I replaced it by that one.

The only thing missing from it seems to be the following logic.
```
        # For int8 compatibility, sometimes the `attn_probs` are in `fp32`
        attn_probs = attn_probs.to(value_states.dtype)
```